components/EnhancedDataTable.tsx

```diff
@@ -92,9 +93,9 @@ function getColumnVisibility<T>(
         .filter((column) => column.name)
         .forEach((column) =>
             colVisProp.push({
-                id: column.name!.toString(),
-                name: column.name!.toString(),
-                visible: columnVisibility[column.name!.toString()],
+                id: getColumnKey(column),
```

Comment:
I know its a super cheap function, but it makes me cringe a bit to see the function called 3 times

Reply:
Good catch! Fixed.